skill-tree /src/main/java/com/RDS/skilltree/Skill/SkillDTO.java

```diff
@@ -0,0 +1,42 @@
+package com.RDS.skilltree.Skill;
+
+import com.RDS.skilltree.User.UserDTO;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.Builder;
+import lombok.Getter;
+
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Getter
+@Builder
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class SkillDTO {
+    private UUID id;
+    private SkillType type;
+    private String name;
+    private Set<UserDTO> users;
+
+    public static SkillDTO toDto(SkillModel skillModel) {
+        return SkillDTO.builder()
+                .id(skillModel.getId())
+                .name(skillModel.getName())
+                .type(skillModel.getType())
+                .build();
+    }
+
+    public static SkillDTO getSkillsWithUsers(SkillModel skillModel) {
+        Set<UserDTO> users = skillModel.getUsers()
+                .stream()
+                .map(UserDTO::toDTO)
+                .collect(Collectors.toSet());
+
+        return SkillDTO.builder()
+                .id(skillModel.getId())
+                .name(skillModel.getName())
+                .type(skillModel.getType())
+                .users(users)
+                .build();
+    }
```

Comment:
Nit: should we keep a separate utils class to have all DRO <-> Entity <-> DTO conversions?

Reply:
We can also use dedicated libraries for this which will reduce the boiler plate code.